docs/components/registrations.md

```diff
@@ -0,0 +1,142 @@
+// instance.
+//  ...
+package server
+
+type endpoint struct {  // (the collected type)
+    ...
+}
+
+type Registration struct {
+    fx.Out
+
+    Endpoint endpoint `group:"server"`
+}
+
+// NewRegistration creates a new Registration instance for the given endpoint.
+func NewRegistration(route string, handler func()) Registration { .. }
+```
+
+Its implementation then requires a slice of the collected type (`endpoint`), again using `group:"server"`:
+
+```go
+// --- comp/server/server.go ---
+
+// endpoint defines an endpoint on this server.
+type endpoint struct { .. }
+
+type dependencies struct {
+    fx.In
+
+    Registrations []endpoint `group:"server"`
+}
+
+func newServer(deps dependencies) Component {
+    // ..
+    for _, e := range deps.Registrations {
+        if e.handler == nil {
+            continue
+        }
+        // ..
+    }
+    // ..
+}
+```
+
+It's good practice to ignore zero values, as that allows providing components to skip
+the registration if desired.
+
+Finally, the providing component (in this case, `foo`) includes a registration in its output as an additional provided type, beyond its `Component` type:
+
+```go
+// --- comp/foo/foo.go ---  (the providing component)
+func newFoo(deps dependencies) (Component, server.Registration) {
+    // ..
+    return foo, server.NewRegistration("/things/foo", foo.handler)
+}
+```
+
+This technique has some caveats to be aware of:
+
+ * The providing components are instantiated before the collecting component.
+ * Fx treats value groups as the collecting component depending on all of the providing components.
+   This means that the providing components cannot depend on the collecting component, as this would represent a dependency cycle.
+ * Fx will instantiate _all_ declared providing components before the collecting component, regardless of whether their `Component` type is required.
+   This may lead to components being instantiated in unexpected circumstances.
+
+## Subscriptions
+
+Subscriptions are a common form of registration, and have support in the `pkg/util/subscriptions` package.
+
+To implement a subscription, the collecting component defines a message type.
+Providing components provide `subscriptions.Subscription[coll.Message]`, from which they can obtain a `subscriptions.Receiver[coll.Message]`.
+Collecting components require `subcriptions.Publisher[coll.Message]`, from which they can obtain a `subscriptions.Transmitter[coll.Message]`.
+
+```go
+// --- announcer/component.go ---
+
+// ...
+// To subscribe to these announcements, provide a subscriptions.Subscription[announcer.Announcement].
+// ...
+package announcer
+```
+
+```go
+// --- announcer/announcer.go ---
+
+func newAnnouncer(pub subscriptions.Publisher[Anouncement]) Component {
+    return &announcer{announcementTx: pub.Transmitter()}  // (get a Transmitter from the Publisher)
+}
+
+// .. later send messages with 
+func (ann *announcer) announce(a announcement) {
```

Comment:
I think it is important to write somewhere in a very explicit way (Maybe that is already the case)  that when using `subscriptions.Subscription[SOMETHING]` SOMETHING has to be a new type and cannot be a raw `string` for example.